File: bindings/python/test_client.py
import sys
import os

# Add build dir to path if needed, or assume installed
# sys.path.append('../../build') 

try:
    import networklib_py
except ImportError:
    print("Error: networklib_py module not found. Make sure it's built and in PYTHONPATH.")
    sys.exit(1)

def main():
    print("Creating EventLoop...")
    loop = networklib_py.EventLoop()
    
    print("Creating Client...")
    client = networklib_py.GrpcClient(loop)
    
    host = "127.0.0.1"
    port = 50051
    
    print(f"Connecting to {host}:{port}...")
    if client.Connect(host, port):
        print("Connected!")
        # Send a test message (raw string for now, proto serialization in python would happen here)
        client.SendUnary("/oms.OrderEntry/Ping", "Hello from Python")
    else:
        print("Connection failed.")

if __name__ == "__main__":
    main()

File: bindings/node/test_client.js
const networklib = require('bindings')('networklib_node');

console.log("Creating EventLoop...");
const loop = new networklib.EventLoop();

console.log("Creating GrpcClient...");
const client = new networklib.GrpcClient(loop);

const host = "127.0.0.1";
const port = 50051;

console.log(`Connecting to ${host}:${port}...`);
if (client.connect(host, port)) {
    console.log("Connected!");
    client.sendUnary("/oms.OrderEntry/Ping", "Hello from Node.js");
} else {
    console.log("Connection failed.");
}

// In a real app, run loop in worker thread or integrate with libuv
// loop.run();
